utils/core/getUserIntentSettings/index.js

```diff
@@ -0,0 +1,120 @@
+import { isUrlMatchRules } from './isUrlMatchRules'
+
+const CTA_TYPE = Object.freeze({
+  SLIDEIN: 'slidein',
+  POPUP: 'popup',
+})
+
+const reduceLatest = (prev, curr) => {
+  if (!prev) return curr
+  const a = new Date(prev.updatedAt)
+  const b = new Date(curr.updatedAt)
+  if (a.getTime() - b.getTime() >= 0) return prev
+  else return curr
+}
+
+function filterLatestUpdateDocument(scenarioDocuments = [], documents = []) {
+  const mapping = documents
+    .map((id) => {
+      const document = scenarioDocuments.find((item) => item.id === id)
+      if (document) {
+        const [ctaType] = document.ctaType.split('_')
+        // console.log("filterLatestUpdateDocument", document.ctaType, ctaType)
+        return {
+          id,
+          updatedAt: document.updatedAt,
+          ctaType,
+        }
+      }
+      return null
+    })
+    .filter((item) => !!item)
+
+  const slideIn = mapping
+    .filter((item) => item.ctaType === CTA_TYPE.SLIDEIN)
+    .reduce(reduceLatest, null)
+  const popup = mapping
+    .filter((item) => item.ctaType === CTA_TYPE.POPUP)
+    .reduce(reduceLatest, null)
+  const result = []
+  if (slideIn) result.push(slideIn)
+  if (popup) result.push(popup)
+  return result
+}
+
+export default function getUserIntentSettings(documents = []) {
+  const initData = {
```

Comment:
setting: {
      pageScroll?:  string | number | boolean,
      pageExit?:  string | number | boolean,
      pageElapsed?:  string | number | boolean,
      pageInactivity?:  string | number | boolean
    }

Reply:
no, this is not the data type response from API. It is the data that is using for tracking event